game/scripts/vscripts/items/meteor_hammer.lua

```diff
@@ -0,0 +1,203 @@
+    ParticleManager:SetParticleControl(self.channel_particle, 0, self:GetCursorPosition())
+    ParticleManager:SetParticleControl(self.channel_particle, 1, Vector(self:GetSpecialValueFor("impact_radius"), 0, 0))
+
+
+
+  end
+end
+
+function item_meteor_hammer:GetChannelAnimation()
+--must implement animation
+return ACT_DOTA_TELEPORT
+end
+
+function item_meteor_hammer:OnChannelFinish(bInterrupted)
+  local caster = self:GetCaster()
+
+  caster:EmitSound("DOTA_Item.MeteorHammer.Cast")
+
+  caster:FadeGesture(ACT_DOTA_TELEPORT)
+
+  if not bInterrupted then
+
+    CreateModifierThinker(caster, self, "modifier_item_meteor_hammer_thinker", {},self:GetCursorPosition(), self:GetCaster():GetTeamNumber(), false)
+
+    end
+  end
+
+function item_meteor_hammer:GetIntrinsicModifierName()
+
+  return "modifier_generic_bonus"
+
+end
+-----------------------------------------------------------------------------------------------------------------------
+-----------------------------------------------------------------------------------------------------------------------
+
+modifier_item_meteor_hammer_thinker = class(ModifierBaseClass)
+
+function modifier_item_meteor_hammer_thinker:OnCreated()
+
+
+  if IsServer() then
+  local ability = self:GetAbility()
+  -- item info from kv
+  self.impact_radius = ability:GetSpecialValueFor("impact_radius")
+  self.impact_damage = ability:GetSpecialValueFor("impact_damage")
+  self.impact_damage_bosses = ability:GetSpecialValueFor("impact_damage_boss")
+
+  self.land_time = ability:GetSpecialValueFor("land_time")
+  self.burn_duration = ability:GetSpecialValueFor("burn_duration")
+  self.stun_duration = ability:GetSpecialValueFor("stun_duration")
+  --landtime should not be a negative number
+  self:StartIntervalThink(self.land_time)
+
+
+
+  end
+end
+
+function modifier_item_meteor_hammer_thinker:OnIntervalThink()
+
+  self:GetParent():EmitSound("DOTA_Item.MeteorHammer.Impact")
+  --PATTACH_ABSORIGIN_FOLLOW, self:GetParent()
+  if IsServer() then
+    self.impact_particle = ParticleManager:CreateParticle("particles/items4_fx/meteor_hammer_spell.vpcf",PATTACH_WORLDORIGIN, nil )
+
+   --Controls the metoer position to origin
+    ParticleManager:SetParticleControl(self.impact_particle, 0, self:GetParent():GetOrigin() + Vector(0, 0, 1000))
+    ParticleManager:SetParticleControl(self.impact_particle, 1, self:GetParent():GetOrigin())
+    --Fade time of cetain particles
+    ParticleManager:SetParticleControl(self.impact_particle, 2, Vector(0.5, 0,0 ) )
+
+    GridNav:DestroyTreesAroundPoint(self:GetParent():GetOrigin(), self.impact_radius, true)
+
+    local ability = self:GetAbility()
+    local enemies = FindUnitsInRadius(self:GetAbility():GetCaster():GetTeamNumber(), self:GetParent():GetOrigin(), self:GetCaster(), self.impact_radius, DOTA_UNIT_TARGET_TEAM_ENEMY, bit.bor(DOTA_UNIT_TARGET_HERO, DOTA_UNIT_TARGET_BASIC), DOTA_UNIT_TARGET_FLAG_NONE , FIND_ANY_ORDER, false)
```

Comment:
`self:GetAbility():GetCaster():GetTeamNubmer()`?

Reply:
That was to get the team the person is on. Is there a better way of doing that?

Reply:
My question is why go through the ability. Use `self:GetCaster()` or better yet, create a local variable for the caster and replace instances of GetCaster with that since it's used a fair number of times in this function.

Reply:
I wasn't sure that I would call the proper reference but with what I know now, that is a reasonable assumption. I'll make the change with the use of a caster local variable.